package Cmd;

import "xCmd.proto";

enum GateParam
{
  GATE_SUPER_USERNUM_CMD = 1;
  GATEPARAM_PUSHOVER_MSG = 2;
  GATEPARAM_ALTER_MSG = 3;
  GATEPARAM_PUSH_TYRANT_DB = 4;
  GATEPARAM_FORWARD_TO_GATEUSER = 5;
}

message GateToSuperUserNum
{
  optional Command cmd = 1 [ default = GATE_SUPER_PROTOCMD ];
  optional GateParam param = 2 [ default = GATE_SUPER_USERNUM_CMD ];

  optional uint32 num = 3 [ default = 0 ];
}

enum EPushMsg
{
  EPUSHMSG_MIN = 0;
  EPUSHMSG_MSG = 1;
  EPUSHMSG_CORE_DUMP = 2;
  EPUSHMSG_SHOP_SELL = 3;
  EPUSHMSG_MESSAGE_STAT = 4;
  EPUSHMSG_TEMP_PACK_FULL = 5;
  EPUSHMSG_MAX = 6;
}
message PushMsgGateSuperCmd
{
  optional Command cmd = 1 [ default = GATE_SUPER_PROTOCMD ];
  optional GateParam param = 2 [ default = GATEPARAM_PUSHOVER_MSG ];

  optional EPushMsg type = 3 [ default = EPUSHMSG_MIN ];
  optional string title = 4;
  optional string msg = 5;
}

message AlterMsgGateSuperCmd
{
  optional Command cmd = 1 [ default = GATE_SUPER_PROTOCMD ];
  optional GateParam param = 2 [ default = GATEPARAM_ALTER_MSG ];

  optional EPushMsg event = 3 [ default = EPUSHMSG_MIN ];
  optional string title = 4;
  optional string msg = 5;
}

message PushTyrantDbGateSuperCmd
{
  optional Command cmd = 1 [ default = GATE_SUPER_PROTOCMD ];
  optional GateParam param = 2 [ default = GATEPARAM_PUSH_TYRANT_DB ];

  optional uint32 accid = 3;
  optional uint64 charid = 4;
  optional string orderid = 5;
  optional uint32 amount = 6;
  optional uint32 itemcount = 7;
  optional string productid = 8;
  optional string chargetype = 9;
}

message ForwardToGateUserCmd
{
  optional Command cmd = 1 [ default = GATE_SUPER_PROTOCMD ];
  optional GateParam param = 2 [ default = GATEPARAM_FORWARD_TO_GATEUSER ];

  repeated uint64 accids = 3;
  optional bytes data = 4;
}
